src/python/grpcio/grpc/_channel.py

```diff
@@ -745,10 +745,10 @@ def __new__(cls, value=_EMPTY_FLAGS):
     def with_wait_for_ready(self, wait_for_ready):
         if wait_for_ready is not None:
             if wait_for_ready:
-                self = self.__class__(self | cygrpc.InitialMetadataFlags.wait_for_ready | \
+                return self.__class__(self | cygrpc.InitialMetadataFlags.wait_for_ready | \
```

Comment:
I know this isn't introduced in this PR, but: why is `_InitialMetadataFlags` extending `int` rather than keeping an int as an instance variable?